java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S3329.html

```diff
@@ -4,54 +4,40 @@
 <p>To generate Initialization Vectors, NIST recommends to use a secure random number generator.</p>
 <h2>Noncompliant Code Example</h2>
 <pre>
-public class MyCbcClass {
+public void encrypt(String key, String plainText) {
+    byte[] bytesIV = "7cVgr5cbdCZVw5WY".getBytes("UTF-8"); // Static initialization vector
 
-  public String applyCBC(String strKey, String plainText) {
-    byte[] bytesIV = "7cVgr5cbdCZVw5WY".getBytes("UTF-8");
-
-    /* KEY + IV setting */
     IvParameterSpec iv = new IvParameterSpec(bytesIV);
     SecretKeySpec skeySpec = new SecretKeySpec(strKey.getBytes("UTF-8"), "AES");
 
-    /* Ciphering */
-    Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+    Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
     cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);  // Noncompliant: the IV is hard coded and thus not generated with a secure random generator
-    byte[] encryptedBytes = cipher.doFinal(plainText.getBytes("UTF-8"));
-    return DatatypeConverter.printBase64Binary(bytesIV)
-            + ";" + DatatypeConverter.printBase64Binary(encryptedBytes);
-  }
```

Comment:
The new code does not compile because `strKey` should be `key`, and the issue is at the wrong location.
I would propose:
```java
  public void encrypt(String key, String plainText) throws GeneralSecurityException {
    byte[] bytesIV = "7cVgr5cbdCZVw5WY".getBytes(StandardCharsets.UTF_8); // Static initialization vector

    IvParameterSpec iv = new IvParameterSpec(bytesIV);  // Noncompliant: the IV is hard coded and thus not generated with a secure random generator
    SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");

    Cipher cipher = Cipher.getInstance("AES/CBC/NoPadding");
    cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
  }
```